src/main.py

```diff
@@ -185,27 +186,40 @@ def _generate_interaction_results(
     results = []
     query_cost = 0.0
     for request in requests:
-        # First, roll out the acting policy.
         task = train_tasks[request.train_task_idx]
-        traj = utils.run_policy_until(
-            request.act_policy,
-            simulator,
-            task.init,
-            request.termination_function,
-            max_num_steps=CFG.max_num_steps_interaction_request)
-        # Now, go through the trajectory and handle queries.
-        # Note: we do this in a second pass so that we can use
-        # the utils.run_policy_until() function called above.
-        # In reality, the agent would query the teacher about
-        # the state only when the environment is in that state.
+        # While transitioning through the environment, when each state is
+        # encountered, query the teacher and record the response.
         responses: List[Optional[Response]] = []
-        for state in traj.states:
-            query = request.query_policy(state)
+
+        # Note that query_policy and responses need to be arguments because
+        # request and responses are changing in the loop.
+        def _process_state(query_policy: Callable[[State], Optional[Query]],
```

Comment:
I'm confused why responses needs to be an argument. Wouldn't it work fine if you just remove this argument, and let the `.append` append directly to the responses variable initialized a few lines ago?

Reply:
I think you're right that this would be logically fine, but pylint complains with `cell-var-in-loop`. so probably leave it, unless you want to add a pylint disable

Reply:
I see. Could we at least rename this variable then? The two `responses` are really different variables.

Reply:
done